### PR TITLE
Patch libcurl configure.ac to work with later versions of autoconf

### PR DESCRIPTION
**Pull request by m-rph**
Adds a patch for autoconf 2.72 for alpine3.20.

Fixes: #144

Target: Custom Docker Image

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by m-rph — COMMENTED** [review]
Diff of "effectively modified segments".

However, we compile libcurl without openssl. So this is effectively no-op.

```diff
4,8c4,8
< if test "${with_secure_transport+set}" = set; then :
<   withval=$with_secure_transport; OPT_SECURETRANSPORT=$withval
<   test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE
< else
<   }Secure-Transport"
---
> if test ${with_secure_transport+y}
> then :
>   withval=$with_secure_transport;
>   OPT_SECURETRANSPORT=$withval
>   test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE, }Secure-Transport"
16,20c16,21
< if test "${with_amissl+set}" = set; then :
<   withval=$with_amissl; OPT_AMISSL=$withval
<   test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE
< else
<   }AmiSSL"
---
> if test ${with_amissl+y}
> then :
>   withval=$with_amissl;
>   OPT_AMISSL=$withval
>   test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE, }AmiSSL"
> 
23a25
> 
28,29c30,33
< if test "${with_ssl+set}" = set; then :
<   withval=$with_ssl; OPT_SSL=$withval
---
> if test ${with_ssl+y}
> then :
>   withval=$with_ssl;
>   OPT_SSL=$withval
32,34c36,38
<     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE
< else
<   }OpenSSL"
---
>     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE, }OpenSSL"
>   fi
> 
37d40
<   fi
41,42c44,47
< if test "${with_openssl+set}" = set; then :
<   withval=$with_openssl; OPT_OPENSSL=$withval
---
> if test ${with_openssl+y}
> then :
>   withval=$with_openssl;
>   OPT_OPENSSL=$withval
44,46c49,51
<     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE
< else
<   }OpenSSL"
---
>     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE, }OpenSSL"
>   fi
> 
49d53
<   fi
54,55c58,61
< if test "${with_gnutls+set}" = set; then :
<   withval=$with_gnutls; OPT_GNUTLS=$withval
---
> if test ${with_gnutls+y}
> then :
>   withval=$with_gnutls;
>   OPT_GNUTLS=$withval
57,59c63,65
<     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE
< else
<   }GnuTLS"
---
>     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE, }GnuTLS"
>   fi
> 
62d67
<   fi
67,68c72,75
< if test "${with_mbedtls+set}" = set; then :
<   withval=$with_mbedtls; OPT_MBEDTLS=$withval
---
> if test ${with_mbedtls+y}
> then :
>   withval=$with_mbedtls;
>   OPT_MBEDTLS=$withval
70,72c77,79
<     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE
< else
<   }mbedTLS"
---
>     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE, }mbedTLS"
>   fi
> 
75d81
<   fi
80,81c86,89
< if test "${with_wolfssl+set}" = set; then :
<   withval=$with_wolfssl; OPT_WOLFSSL=$withval
---
> if test ${with_wolfssl+y}
> then :
>   withval=$with_wolfssl;
>   OPT_WOLFSSL=$withval
83,85c91,93
<     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE
< else
<   }wolfSSL"
---
>     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE, }wolfSSL"
>   fi
> 
88d95
<   fi
93,94c100,103
< if test "${with_bearssl+set}" = set; then :
<   withval=$with_bearssl; OPT_BEARSSL=$withval
---
> if test ${with_bearssl+y}
> then :
>   withval=$with_bearssl;
>   OPT_BEARSSL=$withval
96,98c105,107
<     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE
< else
<   }BearSSL"
---
>     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE, }BearSSL"
>   fi
> 
101d109
<   fi
106,107c114,117
< if test "${with_rustls+set}" = set; then :
<   withval=$with_rustls; OPT_RUSTLS=$withval
---
> if test ${with_rustls+y}
> then :
>   withval=$with_rustls;
>   OPT_RUSTLS=$withval
109,111c119,121
<     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE
< else
<   }rustls"
---
>     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE, }rustls"
>   fi
> 
114d123
<   fi
119,120c128,131
< if test "${with_nss_deprecated+set}" = set; then :
<   withval=$with_nss_deprecated; if test X"$withval" != Xno; then
---
> if test ${with_nss_deprecated+y}
> then :
>   withval=$with_nss_deprecated;
>   if test X"$withval" != Xno; then
130,131c141,144
< if test "${with_nss+set}" = set; then :
<   withval=$with_nss; OPT_NSS=$withval
---
> if test ${with_nss+y}
> then :
>   withval=$with_nss;
>   OPT_NSS=$withval
138,140c151
<     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE
< else
<   }NSS"
---
>     test -z "TLSCHOICE" || TLSCHOICE="${TLSCHOICE:+$TLSCHOICE, }NSS"
168c179,180
< if test "${with_darwinssl+set}" = set; then :
---
> if test ${with_darwinssl+y}
> then :
170a183
> 
```

File: scripts/patch_libcurl_config.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+set -x
+
+(
```
Comment (smirnoal) [inline scripts/patch_libcurl_config.sh:6]: NIT: don't see a strong reason to use ()

**Review by smirnoal — APPROVED** [review]

**Review by m-rph — COMMENTED** [review]
Needs some tidying but otherwise I am satisfied with the end result.

**Comment by m-rph** [comment]
```
pip install https://github.com/aws/aws-lambda-python-runtime-interface-client/archive/refs/heads/python-ric-144.zip
```
Also builds therefore the latest commit is patched properly.

Contents of the tar can be verified by running `scripts/update_deps.sh` and unpacking / diffing the contents.

**Pull request merged**